src/hotspot/os_cpu/linux_x86/os_linux_x86.cpp

```diff
@@ -215,6 +215,13 @@ bool PosixSignals::pd_hotspot_signal_handler(int sig, siginfo_t* info,
   address pc          = NULL;
 
   //%note os_trap_1
+  if (info != NULL && uc != NULL && thread == NULL && sig == SIGSEGV) {
+    pc = (address) os::Posix::ucontext_get_pc(uc);
+    if ((sig == SIGSEGV) && VM_Version::is_cpuinfo_segv_addr(pc)) {
+      // Verify that OS save/restore AVX registers.
+      stub = VM_Version::cpuinfo_cont_addr();
+    }
+  } else
```

Comment:
Why do we need this block? Is not this duplicating logic from line 251?

Reply:
It was needed for some previous variant of the patch and it is a leftover. Removed. Sorry for not self-reviewing this version of my patch.

Reply:
It is in fact needed: https://github.com/openjdk/crac/blob/08c2c0f2b140b1011fec39a1a83037df8e59577d/src/hotspot/os_cpu/linux_x86/os_linux_x86.cpp#L217
During CRaC restore there is somehow no Java thread created yet that time. I do not understand this much so I left it there as working but maybe some Java thread should be initialized earlier?